src/Microsoft.OData.Core/Uri/ApplyClauseToStringBuilder.cs

```diff
@@ -148,6 +154,45 @@ private void Translate(ComputeTransformationNode transformation)
             }
         }
 
+        private void Translate(ExpandTransformationNode transformation)
+        {
+            ExpandedNavigationSelectItem expandedNavigation = transformation.ExpandClause.SelectedItems.Single() as ExpandedNavigationSelectItem;
+            AppendExpandExpression(expandedNavigation);
+        }
+
+        private void AppendExpandExpression(ExpandedNavigationSelectItem expandedNavigation)
+        {
+            AppendExpression(expandedNavigation.PathToNavigationProperty);
+
+            // Append filter
+            if (expandedNavigation.FilterOption != null)
+            {
+                AppendComma(true);
+                if (expandedNavigation.FilterOption != null)

```

Comment:
Can get rid of this inner check now, right? Since you broke out the SelectAndExpand case it's redundant with the outer conditional.